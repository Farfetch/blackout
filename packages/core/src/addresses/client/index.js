/**
 * Addresses clients.
 *
 * @module addresses/client
 * @category Addresses
 * @subcategory Clients
 */

export { default as getPredictions } from './getPredictions';
export { default as getPredictionDetails } from './getPredictionDetails';

export { default as getAddresses } from './getAddresses';
export { default as getAddress } from './getAddress';
export { default as postAddress } from './postAddress';
export { default as putAddress } from './putAddress';
export { default as deleteAddress } from './deleteAddress';
export { default as putDefaultBillingAddress } from './putDefaultBillingAddress';
export { default as putDefaultShippingAddress } from './putDefaultShippingAddress';
export { default as getSchema } from './getSchema';
export { default as putDefaultContactAddress } from './putDefaultContactAddress';
export { default as deleteDefaultContactAddress } from './deleteDefaultContactAddress';
export { default as getDefaultContactAddress } from './getDefaultContactAddress';
