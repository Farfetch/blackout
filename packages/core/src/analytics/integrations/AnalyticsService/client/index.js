/**
 * @module AnalyticsService/client
 * @category Analytics
 * @subcategory Clients
 */

/**
 * Trackings client.
 */
export { default as postAnalytics } from './postAnalytics';
