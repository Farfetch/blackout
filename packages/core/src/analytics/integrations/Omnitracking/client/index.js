/**
 * @module Omnitracking/client
 * @category Analytics
 * @subcategory Clients
 */

/**
 * Trackings client.
 */
export { default as postTrackings } from './postTrackings';
export { default as postBatchTrackings } from './postBatchTrackings';
export * from './postTrackings';
