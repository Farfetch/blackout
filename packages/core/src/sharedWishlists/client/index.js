/**
 * Shared wishlists clients.
 *
 * @module sharedWishlist/client
 * @category Shared Wishlist
 * @subcategory Clients
 */

export { default as getSharedWishlist } from './getSharedWishlist';
export { default as deleteSharedWishlist } from './deleteSharedWishlist';
export { default as postSharedWishlist } from './postSharedWishlist';
export { default as putSharedWishlist } from './putSharedWishlist';
