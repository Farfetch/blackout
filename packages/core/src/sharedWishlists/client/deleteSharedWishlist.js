import client, { adaptError } from '../../helpers/client';
import join from 'proper-url-join';

/**
 * Method responsible for deleting a shared wishlist.
 *
 * @function deleteSharedWishlist
 * @memberof module:sharedWishlists/client
 *
 * @param {string} id - Universal identifier of the shared wishlist.
 * @param {object} [config] - Custom configurations to send to the client
 * instance (axios).
 *
 * @returns {Promise} Promise that will be resolved when the call to the
 * endpoint finishes.
 */
export default (id, config) =>
  client
    .delete(join('/commerce/v1/sharedWishlists', id), config)
    .catch(error => {
      throw adaptError(error);
    });
