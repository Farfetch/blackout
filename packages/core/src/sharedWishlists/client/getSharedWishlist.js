import client, { adaptError } from '../../helpers/client';
import join from 'proper-url-join';

/**
 * Method responsible for loading the shared wishlist.
 *
 * @function getSharedWishlist
 * @memberof module:sharedWishlists/client
 *
 * @param {string} id - Universal identifier of the shared wishlist.
 * @param {object} [config] - Custom configurations to send to the client
 * instance (axios).
 *
 * @returns {Promise} Promise that will be resolved when the call to the
 * endpoint finishes.
 */
export default (id, config) =>
  client
    .get(join('/commerce/v1/sharedWishlists', id), config)
    .then(response => response.data)
    .catch(error => {
      throw adaptError(error);
    });
