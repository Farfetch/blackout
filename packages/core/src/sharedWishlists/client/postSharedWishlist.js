import client, { adaptError } from '../../helpers/client';

/**
 * @typedef {object} PostSharedWishlistData
 *
 * @alias PostSharedWishlistData
 * @memberof module:sharedWishlists/client
 *
 * @property {string} id - Universal identifier of the shared wishlist.
 * @property {string} setId - Global identifier of the set to retrieve information
 * from.
 */

/**
 * Method responsible for creating a snapshot of the wishlist set to allow share it.
 *
 * @function postSharedWishlist
 * @memberof module:sharedWishlists/client
 *
 * @param {PostSharedWishlistData} data - Details of the information of which wishlist
 * set is going to be shared.
 * @param {object} [config] - Custom configurations to send to the client
 * instance (axios).
 *
 * @returns {Promise} Promise that will be resolved when the call to
 * the endpoint finishes.
 */
export default (data, config) =>
  client
    .post('/commerce/v1/sharedWishlists', data, config)
    .then(response => response.data)
    .catch(error => {
      throw adaptError(error);
    });
