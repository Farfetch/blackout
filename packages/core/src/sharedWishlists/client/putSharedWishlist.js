import client, { adaptError } from '../../helpers/client';
import join from 'proper-url-join';

/**
 * Method responsible for updating a shared wishlist, syncronizing the prior
 * snapshot with the current state of the related wishlist set.
 *
 * @function putSharedWishlist
 * @memberof module:sharedWishlists/client
 *
 * @param {string} id - Universal identifier of the shared wishlist.
 * @param {object} [config] - Custom configurations to send to the client instance (axios).
 *
 * @returns {Promise} Promise that will be resolved when the call to the endpoint finishes.
 */
export default (id, config) =>
  client
    .put(join('/commerce/v1/sharedWishlists', id), config)
    .then(response => response.data)
    .catch(error => {
      throw adaptError(error);
    });
