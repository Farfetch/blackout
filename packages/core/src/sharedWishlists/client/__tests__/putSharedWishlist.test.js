import {
  mockSharedWishlistId,
  mockSharedWishlistsResponse,
} from 'tests/__fixtures__/sharedWishlists';
import { putSharedWishlist } from '..';
import client from '../../../helpers/client';
import fixtures from '../__fixtures__/putSharedWishlist.fixtures';
import moxios from 'moxios';

describe('putSharedWishlist', () => {
  const expectedConfig = undefined;
  const sharedWishlistId = mockSharedWishlistId;
  const spy = jest.spyOn(client, 'put');

  beforeEach(() => {
    moxios.install(client);
    jest.clearAllMocks();
  });

  afterEach(() => moxios.uninstall(client));

  it('should handle a client request successfully', async () => {
    const response = mockSharedWishlistsResponse;

    fixtures.success({ sharedWishlistId, response });

    await expect(putSharedWishlist(sharedWishlistId)).resolves.toBe(response);

    expect(spy).toHaveBeenCalledWith(
      `/commerce/v1/sharedWishlists/${sharedWishlistId}`,
      expectedConfig,
    );
  });

  it('should receive client request error', async () => {
    fixtures.failure({ sharedWishlistId });

    await expect(putSharedWishlist(sharedWishlistId)).rejects.toMatchSnapshot();

    expect(spy).toHaveBeenCalledWith(
      `/commerce/v1/sharedWishlists/${sharedWishlistId}`,
      expectedConfig,
    );
  });
});
