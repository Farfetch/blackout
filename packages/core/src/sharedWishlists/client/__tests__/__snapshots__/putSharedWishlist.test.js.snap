// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`putSharedWishlist should receive client request error 1`] = `
Object {
  "code": -1,
  "message": "stub error",
  "status": 404,
}
`;
